src/Adapter/Category/CategoryRemover.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Category;
+
+use Category;
+use PrestaShop\PrestaShop\Core\Domain\Product\Category\ValueObject\CategoryDeletionMode;
+use PrestaShop\PrestaShop\Core\Foundation\Database\DatabaseInterface;
+use Product;
+use Shop;
+use Symfony\Component\Translation\TranslatorInterface;
+use Validate;
+
+/**
+ * Class CategoryRemover is responsible for deleting legacy categories.
+ *
+ * @internal
+ */
+class CategoryRemover
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var DatabaseInterface
+     */
+    private $database;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param DatabaseInterface $database
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        DatabaseInterface $database
+    ) {
+        $this->translator = $translator;
+        $this->database = $database;
+    }
+
+    /**
+     * Delete single category.
+     *
+     * @param int $categoryId
+     * @param CategoryDeletionMode $mode
+     *
+     * @return string[] Errors if any
+     */
+    public function remove($categoryId, CategoryDeletionMode $mode)
+    {
+        $errors = [];
+
+        $category = new Category($categoryId);
+
+        if (!Validate::isLoadedObject($category)) {
```

Comment:
use ValidateAdapter maybe?

Reply:
is it really worth it? actually, ill remove it and use command for deleting category.

Reply:
@mickaelandrieu said me to don't use old classes if there is an adapter for that =)